src/main/java/com/artipie/rpm/FileChecksum.java

```diff
@@ -76,7 +77,7 @@ public String hex() throws IOException {
         try (FileChannel chan = FileChannel.open(this.file, StandardOpenOption.READ)) {
             final ByteBuffer buf = ByteBuffer.allocateDirect(FileChecksum.BUF_SIZE);
             while (chan.read(buf) > 0) {
-                buf.flip();
+                ((Buffer) buf).flip();
```

Comment:
@g4s8 Couldn't we avoid to use this casting here?

Reply:
@paulodamaso no, there are some issues with `flip()` method on different Java version: on JDK 1.8 `flip` is a method of `Buffer` class which is superclass of `ByteBuffer` and it's `Buffer flip()`. But on JDK > 1.8 `ByteBuffer` changed return type of this method to `ByteBuffer flip()`. So we may get an `NoSuchMethodError` because of that if we compile this module with `1.8` jdk. Just google this error to understand its popularity: "bytebuffer flip nosuchmethoderror"